tests/components/pooldose/test_number.py

```diff
@@ -112,3 +113,63 @@ async def test_set_number_value(
     # Verify state updated immediately (optimistic update)
     ph_target_state = hass.states.get("number.pool_device_ph_target")
     assert ph_target_state.state == "7.0"
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_actions_cannot_connect_number(
+    hass: HomeAssistant,
+    mock_pooldose_client: AsyncMock,
+    init_integration: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """When the client write method raises, ServiceValidationError('cannot_connect') is raised."""
+    client = mock_pooldose_client
+    entity_id = "number.pool_device_ph_target"
+    before = hass.states.get(entity_id)
+    assert before is not None
+
+    err = ServiceValidationError(
+        translation_domain="pooldose",
+        translation_key="cannot_connect",
+        translation_placeholders={"error": "mocked error"},
+    )
+
+    client.set_number = AsyncMock(side_effect=err)
```

Comment:
This is incorrect. the `client.set_number` doesn't raise `ServiceValidationError`, it just returns a falsey state and the new exception handler does the rest. So instead we should mimick what the library would've done